cpp/tests/utilities/base_fixture.hpp

```diff
@@ -167,5 +164,53 @@ class TempDirTestEnvironment : public ::testing::Environment {
   }
 };
 
+class RmmTestEnvironment : public ::testing::Environment {
+ std::string mode;
+ public:
+  RmmTestEnvironment(std::string const& mode):mode(mode){}
+  void SetUp() {
+    rmmOptions_t opts{};
+    if (mode == "cuda")
+      opts.allocation_mode = CudaDefaultAllocation;
+    else if (mode == "pool")
+      opts.allocation_mode = PoolAllocation;
+    else if (mode == "managed")
+      opts.allocation_mode = CudaManagedMemory;
+    else 
+      CUDF_FAIL("Invalid RMM allocation mode");
+
+    ASSERT_EQ(rmmInitialize(&opts), RMM_SUCCESS);
+  }
+
+  void TearDown() {
+    ASSERT_EQ(rmmFinalize(), RMM_SUCCESS);
+  }
+};
+
 }  // namespace test
 }  // namespace cudf
+
+inline auto parse_cudf_opts(int argc, char **argv) {
+  try {
+    cxxopts::Options options(argv[0], " - cuDF tests command line options");
+    options
+      .allow_unrecognised_options()
+      .add_options()
+      ("rmm_mode", "RMM allocation mode", cxxopts::value<std::string>()->default_value("pool"))
+    ;
+
+    return options.parse(argc, argv);
+  }
+  catch (const cxxopts::OptionException& e) {
+    CUDF_FAIL("Error parsing command line options");
+  }
+}
+
+#define CUDF_TEST_PROGRAM_MAIN() int main(int argc, char **argv) { \
```

Comment:
Can you add some docs for the new environment? Especially the part about needing to use the `CUDF_TEST_PROGRAM_MAIN` in all test files now.